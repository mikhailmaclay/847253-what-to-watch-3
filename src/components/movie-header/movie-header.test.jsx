// Libraries
import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
// Components
import MovieHeader from './movie-header';

describe(`<MovieHeader/>`, () => {
  const props = {
    name: `The Grand Budapest Hotel`,
    background: `/img/movies/backgrounds/1.jpg`,
    genre: `Drama`,
    releaseDate: `March 13, 2014`
  };

  it(`should render correctly`, () => {
    const result = renderer.create(
        <BrowserRouter>
          <MovieHeader {...props}/>
        </BrowserRouter>
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});
