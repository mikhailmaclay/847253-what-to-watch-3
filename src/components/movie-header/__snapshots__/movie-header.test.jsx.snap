// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieHeader/> Should render correctly 1`] = `
<div
  className="movie-card__hero"
>
  <div
    className="movie-card__bg"
    style={
      Object {
        "backgroundColor": "black",
      }
    }
  >
    <img
      alt="The Grand Budapest Hotel"
      src="/img/movies/backgrounds/1.jpg"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href={null}
        onClick={null}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/sign-in"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__desc"
    >
      <h2
        className="movie-card__title"
      >
        The Grand Budapest Hotel
      </h2>
      <p
        className="movie-card__meta"
      >
        <span
          className="movie-card__genre"
        >
          Drama
        </span>
        <span
          className="movie-card__year"
        >
          March 13, 2014
        </span>
      </p>
      <div
        className="movie-card__buttons"
      >
        <a
          className="btn btn--play movie-card__button"
          href="/watch/1"
          onClick={[Function]}
        >
          <svg
            height="19"
            viewBox="0 0 19 19"
            width="19"
          >
            <use
              xlinkHref="#play-s"
            />
          </svg>
          <span>
            Play
          </span>
        </a>
        <button
          className="btn btn--list movie-card__button"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="20"
            viewBox="0 0 19 20"
            width="19"
          >
            <use
              xlinkHref="#in-list"
            />
          </svg>
          <span>
            My list
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
